src/pydicom/fileset.py

```diff
@@ -103,10 +108,8 @@ def generate_filename(
     if len(prefix) > 7:
         raise ValueError("The 'prefix' must be less than 8 characters long")
 
-    chars = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ"
-    if not alphanumeric:
-        chars = chars[:10]
-
+    chars = "ABCDEFGHIJKLMNOPQRSTUVWXYZ" if use_alpha else "0123456789"
```

Comment:
I probably would have just stripped the last letter (or maybe omit J instead of Z, as this is sometimes omitted as it is too similar to I) and leave the digits in, as these are the commonly used first ID characters (and leave the argument name as is).

Reply:
Done